### DISCUSSION TITLE
Available streams only two

### DISCUSSION THREAD

**Discussion by Yuji0531** [streamlink/streamlink#4694] [Q&A] [answered]
Available streams: live (worst, best)

I used best in "https://dwm2r7v7sxy49.cloudfront.net/live01/live_1m_20220731T142835.m3u8?Expires=1659338762&Signature=WQjDV2NzE7-F787euzbqEmKLNdHSQRwaqqwR2QsYyJoMzdRiGDXfwRzmvAt2KKzinAQnkFwkiX6KHvP9qEVZYOrbC2eNOUO5uln8gDbPhLZ~zPIT3yU0Bwb-V8g10aBD~TcXUfV~hTw~NPTemhhuatQkjYDllpOVbQ8ilzwrSLsz0ZBGbZiRc9nLqvOiCYXdhf9Q0sE4v4PuCPhBiYJxgkuHs95FPmSlvVI8olCPs6MYeAEeYj59GdoVxUIxN1lITSTbS3Qu0efI3DMyNAJySpyb~qZt2QVbvoDXcGDUdgZbX-aL64v~~crPzAXBJfs26L~DtngROyxxas2wRcqXTw__&Key-Pair-Id=APKAICPKDW5EZGX4OD4A"
I got 960 * 540 Video. But I could watch stream 1920 * 1080.
Why the best option is not best?

**Comment by bastimeyer** [comment, marked as answer]
This is a **media playlist**, not a **multivariant playlist**, so there's only **one stream**. The HLS plugin yields media playlists as "live" ("worst" and "best" being the same):
https://github.com/streamlink/streamlink/blob/4.2.0/src/streamlink/plugins/hls.py#L25-L27

**Reply by Yuji0531** [reply]
Is it possible to force it to 1080p?

**Reply by bastimeyer** [reply]
You need to use a different URL then. You are using a specific **media playlist** URL for this **specific quality**.

**Reply by Yuji0531** [reply]
Oh,ok.
You mean the m3u8 URL I found in the developer tools had this image quality?
I will try next time to see if there are other image qualities.

**Reply by bastimeyer** [reply]
https://datatracker.ietf.org/doc/html/draft-pantos-hls-rfc8216bis#section-2

> A Playlist is either a Media Playlist or a Multivariant Playlist.

> A Media Playlist contains a list of Media Segments, which, when played sequentially, will play the multimedia presentation.

> A more complex presentation can be described by a Multivariant Playlist.  A Multivariant Playlist provides a set of Variant Streams, each of which describes a different version of the same content.
> A Variant Stream includes a Media Playlist that specifies media encoded at a particular bit rate, in a particular format, and at a particular resolution for media containing video.
> A Variant Stream can also specify a set of Renditions.  Renditions are alternate versions of the content, such as audio produced in different languages or video recorded from different camera angles.

**Reply by Yuji0531** [reply]
I couldn't find a Multivariant Playlist... Any tips on how to find them on network tabs, etc.?

**Reply by bastimeyer** [reply]
You'll have to figure this out yourself.
